classes/condition.php

```diff
@@ -23,163 +27,135 @@
 namespace availability_shibboleth2fa;
 
 use availability_shibboleth2fa\event\user_2fa_loggedin;
+use coding_exception;
+use context_course;
+use core\event\course_deleted;
+use core\event\user_deleted;
+use core\event\user_enrolment_deleted;
+use core_availability\condition as abstract_condition;
 use core_availability\info;
 use core_availability\info_module;
 use core_availability\info_section;
+use dml_exception;
+use html_writer;
+use moodle_exception;
+use moodle_url;
+use stdClass;
 
-defined('MOODLE_INTERNAL') || die();
```

Comment:
Because it is unnecessary, when there are no side effects present. Pure class definition files like the `condition` here are safe and that check is therefore unneeded. The Moodle Code Sniffer called by the Moodle Plugin CI even [picks up](https://github.com/moodlehq/moodle-cs/blob/89ff0acd727f8611cbafe5aea524614090b24ad9/moodle/Sniffs/Files/MoodleInternalSniff.php#L131-L142) these unnecessary "guards" and emits a warning.